pkg/kubelet/certificate/certificate_manager.go

```diff
@@ -186,18 +187,55 @@ func (m *manager) Start() {
 
 	glog.V(2).Infof("Certificate rotation is enabled.")
 
-	err := m.rotateCerts()
-	if err != nil {
-		glog.Errorf("Could not rotate certificates: %v", err)
+	m.setRotationDeadline()
```

Comment:
If the kubelet gets restarted, and setRotationDeadline() gets called again, it will recalculate the deadline for the currently selected certificate. The deadline is kept in memory so it's lost on a restart. But the certificate has all the data needed to completely recalculate it.